skrf/frequency.py

```diff
@@ -318,7 +319,14 @@ def npoints(self, n):
         '''
         set the number of points in the frequency
         '''
-        self.f = linspace(self.start, self.stop, n)
+
+        if self.sweep_type == 'lin':
+            self.f = linspace(self.start, self.stop, n)
```

Comment:
should we used 'linear' (like in matplotlib) or just 'lin' as proposed ? I'm more in favor in 'linear' because I'm used to matplotlib, but tbd

Reply:
I would prefer 'linear' too, however I didn't want to introduce a breaking change.
Personally I prefer Enums instead of strings in general.